scan-command/src/main/java/io/ballerina/scan/internal/ProjectAnalyzer.java

```diff
@@ -69,51 +97,119 @@ private Consumer<DocumentId> analyzeDocument(Module module, ScannerContextImpl s
+            CompilerPluginDescriptor pluginDesc = pkgManifest.compilerPluginDescriptor().get();
+            List<String> jarPaths = new ArrayList<>();
+            pluginDesc.dependencies().forEach(dependency -> {
+                jarPaths.add(dependency.getPath());
+            });
 
-        List<Issue> externalIssues = new ArrayList<>(scannerContextList.size());
-        for (ScannerContext scannerContext : scannerContextList) {
-            ReporterImpl reporter = (ReporterImpl) scannerContext.getReporter();
-            externalIssues.addAll(reporter.getIssues());
+            List<URL> jarUrls = new ArrayList<>();
+            jarPaths.forEach(jarPath -> {
+                try {
+                    URL jarUrl = Path.of(jarPath).toUri().toURL();
+                    jarUrls.add(jarUrl);
+                } catch (MalformedURLException ex) {
+                    throw new RuntimeException(ex);
+                }
+            });
+
+            URLClassLoader ucl = AccessController.doPrivileged((PrivilegedAction<URLClassLoader>) () ->
+                    new URLClassLoader(jarUrls.toArray(new URL[0]), this.getClass().getClassLoader())
+            );
+            InputStream resourceAsStream = ucl.getResourceAsStream(RULES_FILE);
+            if (resourceAsStream == null) {
+                continue;
+            }
+
+            StringBuilder stringBuilder = new StringBuilder();
+            try (BufferedReader reader = new BufferedReader(
+                    new InputStreamReader(resourceAsStream,
+                            StandardCharsets.UTF_8))) {
+                String line;
+                while ((line = reader.readLine()) != null) {
+                    stringBuilder.append(line);
+                }
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+
+            String fqn = pluginDesc.plugin().getClassName();
+            List<Rule> externalRules = new ArrayList<>();
+            Gson gson = new Gson();
+            JsonArray ruleArray = gson.fromJson(stringBuilder.toString(), JsonArray.class);
+            ruleArray.forEach(rule -> {
+                JsonObject ruleObject = rule.getAsJsonObject();
+                int numericId = ruleObject.get("id").getAsInt();
+                RuleKind ruleKind = switch (ruleObject.get("ruleKind").getAsString()) {
+                    case "BUG" -> RuleKind.BUG;
+                    case "VULNERABILITY" -> RuleKind.VULNERABILITY;
+                    case "CODE_SMELL" -> RuleKind.CODE_SMELL;
+                    default -> null;
+                };
+
+                String description = ruleObject.get("description").getAsString();
+                if (ruleKind != null) {
+                    Rule inMemoryRule = RuleFactory.createRule(numericId, description, ruleKind,
+                            org.value(), name.value());
+                    externalRules.add(inMemoryRule);
+                }
```

Comment:
Given you use `numericId` and `description` only if `ruleKind` is not null I think you can restrcuture things to be more concise  by moving them closer to where you are actually using them
```suggestion
                                RuleKind ruleKind = switch (ruleObject.get("ruleKind").getAsString()) {
                    case "BUG" -> RuleKind.BUG;
                    case "VULNERABILITY" -> RuleKind.VULNERABILITY;
                    case "CODE_SMELL" -> RuleKind.CODE_SMELL;
                    default -> null;
                };

                if (ruleKind != null) {
                    int numericId = ruleObject.get("id").getAsInt();
                    String description = ruleObject.get("description").getAsString();
                    externalRules.add(RuleFactory.createRule(numericId, description, ruleKind,
                            org.value(), name.value()));
                }

```